falco.yaml

```diff
@@ -460,87 +460,106 @@ metadata_download:
-# even when choosing this option. This option allows you to define a
-# set of base syscalls that will be activated in addition to the
-# syscalls defined in the rules.
-#
-# You may ask yourself why do we need to activate syscalls in addition to the rules?
-#
-# Falco requires a set of syscalls to build up state in userspace. This is because for
-# example when spawning a new process or creating a network connection more than one syscall
-# is involved. Furthermore, properties of a process during its life time can be modified
-# by syscalls. Falco takes care of this by activating more syscalls than the ones defined
-# in the rules and by managing a smart process cache table in userspace.
-# Processes are purged when a process exits.
-#
-# Looking back to what this option does, it activates all syscalls from the rules
-# (including resolved macros) and the ones specified here.
-#
-# This puts the end user in the driver seat to tell Falco what it needs, but if not used correctly
-# Falco logs may be incomplete or wrong or Falco won't work at all. This option however can be
-# very useful to lower CPU utilization and allowing you to tailor Falco to specific environments
-# according to your organization's threat model and cost budget.
-#
-# !!! When NOT using this option, Falco defaults to adding a static (more verbose) set of syscalls
-# in addition to the rules system calls Falco needs for its state engine build-up and life-cycle management.
-#
-# `base_syscalls.repair` is an experimental alternative to Falco's default state engine enforcement.
-# `base_syscalls.repair` is designed to be the most resourceful option to ensure Falco runs correctly
-# while activating a most minimal set of additional syscalls. The recommendations listed in the suggestions
-# section is effectively what `base_syscalls.repair` is doing in an automated manner. `base_syscalls.repair`
-# can be used with an empty custom set.
+# This option configures the set of syscalls that Falco traces.
+#
+# --- [Falco's State Engine]
+#
+# Falco requires a set of syscalls to build up state in userspace.
+# For example, when spawning a new process or network connection, more than one
+# syscall is involved.
+# Furthermore, properties of a process during its lifetime can be modified by
+# syscalls. Falco accounts for this by activating additional syscalls than the
+# ones defined in the rules and by managing a smart process cache table in
+# userspace. Processes are purged from this table when a process exits.
+#
+# By default, with
+#   ```
+#   base_syscalls.custom_set = []
+#   base_syscalls.repair = false
+#   ```
+# Falco enables tracing for syscalls:
+#   (1) in (enabled) Falco rules
+#   (2) in a static, more verbose set defined in `libsinsp::events::sinsp_state_sc_set` in libs/userspace/libsinsp/events/sinsp_events_ppm_sc.cpp
+# This allows Falco to successfully build up it's state engine and life-cycle management.
+#
```

Comment:
Maybe we need a segue here as to why this option exists. Something similar to: "If the default option does not fit the Falco use case this `base_syscalls` option provides full end user control around the total set of syscalls that will be activated" -> needs some re-phrasing.

Reply:
How does the following sound?

```
If the default behavior described above does not fit the user's use case for Falco, the `base_syscalls.custom_set` option allows for finer end-user control of syscalls traced by Falco.
```

And that directly flows into the section describing `base_syscalls.custom_set`

Reply:
This sounds great perhaps even only say ... for Falco, the `base_syscalls` option allows ...

Reply:
👍

```
# If the default behavior described above does not fit the user's use case for Falco,
# the `base_syscalls` option allows for finer end-user control of syscalls traced by Falco.
```